pyurtrcon.py

```diff
@@ -1,119 +1,164 @@
+
+        try:
+            import readline
+        except:
+            self.intro += '\n(!) "readline" isn\'t installed. Autocomplete unavailable!'
+
+    def do_gameserver(self, args):
+        """Set or display the gameserver address.\nSupply the address in host:port format to set it."""
+        if args == '':
+            print('Address of the gameserver is "{}:{}"'.format(self.server_host, self.server_port))
+        else:
+            self.server_host = args.split(':')[0]
+            self.server_port = int(args.split(':')[1])
+
+    def do_password(self, args):
+        """Set or display the RCon password."""
+        if args == '':
+            print('RCon password is "{}"'.format(self.rcon_password))
+        else:
+            self.rcon_password = args
+
+    def do_rcon(self, args):
+        """Send RCon commands to UrT server."""
+        try:
+            socket_handle = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            socket_handle.connect((self.server_host, self.server_port))
+            socket_handle.settimeout(self.timeout)
+            socket_handle.send('{}rcon {} {}'.format(self.magic, self.rcon_password, args))
+            (response, address) = socket_handle.recvfrom(1024)
+            socket_handle.close()
+            formatted_response = re.sub(r'\^.', '', response.split('\n')[1])
+            if len(formatted_response):
+                print(formatted_response)
+        except Exception as e:
+            print('Error! Details: {}'.format(e.message))
+
+    def complete_rcon(self, primitive, *_):
+        if primitive:
+            return [ command for command in self.common_commands if command.startswith(primitive) ]
+        else:
+            return self.common_addresses
+
+    def do_history(self, _):
+        """Prints all the previous commands of this session."""
+        for entry in self._hist:
+            print(entry)
+
+    def do_help(self, args):
+        """'help' or '?' with no arguments will print all commands.\n'help <command>' or '? <command>' will print help on <command>."""
+        cmd.Cmd.do_help(self, args)
+
+    def do_exit(self, _):
+        """Exits from the PyUrTRCon tool."""
+        return -1
+
+    def do_EOF(self, _):
+        return -1
+
+    def preloop(self):
+        self._hist  = []
+        self._locals  = {}
+        self._globals = {}
+        cmd.Cmd.preloop(self)
+
+    def postloop(self):
+        cmd.Cmd.postloop(self)
+
+    def precmd(self, line):
+        self._hist += [ line.strip() ]
+        return line
+
+    def postcmd(self, stop, line):
+        return stop
+
+    def emptyline(self):   
+        pass
+
+    def default(self, line):      
+        print('Unrecognized command. Type "help" for instructions.')
+
+    undoc_header = None
+
+    def print_topics(self, header, cmds, cmdlen, maxcol):
+        if header is not None:
+            cmd.Cmd.print_topics(self, header, cmds, cmdlen, maxcol)
+
+
+class GistAuth(object):
```

Comment:
this is pretty cool, but can you explain what the reasoning is for auth via contents of gist?